faq-bundle/contao/modules/ModuleFaqList.php

```diff
@@ -140,18 +140,28 @@ protected function generateFaqLink($objFaq)
 			throw new \Exception("FAQ categories without redirect page cannot be used in an FAQ list");
 		}
 
-		// Get the URL from the jumpTo page of the category
-		if (!isset($this->arrTargets[$jumpTo]))
+		if ($jumpTo && ($objTarget = $this->getPageWithDetails($jumpTo)) !== null)
 		{
-			$this->arrTargets[$jumpTo] = StringUtil::ampersand(Environment::get('requestUri'));
+			/** @var PageModel $objTarget */
+			return StringUtil::ampersand($objTarget->getFrontendUrl('/' . ($objFaq->alias ?: $objFaq->id)));
+		}
+
+		return StringUtil::ampersand(Environment::get('requestUri'));
+	}
 
-			if ($jumpTo > 0 && ($objTarget = PageModel::findByPk($jumpTo)) !== null)
-			{
-				/** @var PageModel $objTarget */
-				$this->arrTargets[$jumpTo] = StringUtil::ampersand($objTarget->getFrontendUrl('/%s'));
-			}
+	/**
+	 * Return the page object with loaded details for the given page ID
+	 *
+	 * @param  integer        $intPageId
+	 * @return PageModel|null
+	 */
+	private function getPageWithDetails($intPageId)
+	{
+		if (!isset(self::$arrPageCache[$intPageId]))
```

Comment:
this would return false if the page model was not found in the registry. Maybe we should use `array_key_exists` instead (which would yield true in case of `null` value afaik)?

Reply:
Done in fed7a4d874d34e2f4bde601d04432c49e4b61775